datalad/support/exceptions.py

```diff
@@ -174,6 +174,17 @@ def __str__(self):
                "".format(self.remote, linesep, super_str)
 
 
+class IncompleteResultsError(RuntimeError):
+    """Exception to be raised whenever results are incomplete.
+
+    Any results produced nevertheless are to be passed as `results`,
+    and become available via the `results` attribute.
+    """
+    def __init__(self, results, **kwargs):
+        super(IncompleteResultsError, self).__init__(**kwargs)
```

Comment:
since "message" goes as the first positional argument to RuntimeError (and thus RuntimeError), might be worth adding and passing it explicitly here as well `(message=None, results=None, **kwargs)`